src/components/GenericTable/ColumnVisibilityPopper.tsx

```diff
@@ -0,0 +1,320 @@
+  },
+}));
+
+const StyledCloseDialogButton = styled(IconButton)(() => ({
+  position: "absolute !important" as "absolute",
+  right: "21px",
+  top: "11px",
+  padding: "10px",
+  "& svg": {
+    color: "#44627C",
+  },
+}));
+
+const UncheckedIcon = styled("div")({
+  outline: "3px solid #1D91AB",
+  outlineOffset: -3,
+  borderRadius: "4px",
+  width: "24px",
+  height: "24px",
+  backgroundColor: "#FFFFFF",
+  color: "#083A50",
+  cursor: "pointer",
+});
+
+const CheckedIcon = styled("div")({
+  backgroundImage: `url(${checkboxCheckedIcon})`,
+  backgroundSize: "auto",
+  backgroundRepeat: "no-repeat",
+  width: "24px",
+  height: "24px",
+  backgroundColor: "initial",
+  color: "#1D91AB",
+  borderRadius: "4px",
+  cursor: "pointer",
+});
+
+const StyledTitle = styled(Typography)({
+  fontSize: "16px",
+  color: "#083A50 !important",
+  fontFamily: "'Nunito', 'Rubik', sans-serif",
+  fontStyle: "normal",
+  fontWeight: 700,
+  lineHeight: "19.6px",
+});
+
+const StyledCheckbox = styled(Checkbox)({
+  padding: 0,
+  marginRight: "5px",
+});
+
+const StyledPaper = styled(Paper)({
+  width: "295px",
+  border: "2px solid #5AB8FF",
+  borderRadius: "8px",
+  backgroundColor: "#F2F6FA",
+});
+
+const StyledColumnList = styled(Box)({
+  overflowY: "auto",
+  display: "flex",
+  flexDirection: "column",
+  justifyContent: "center",
+  alignItems: "flex-start",
+  gap: "10px",
+  marginBottom: "12px",
+  padding: "22px 16px 18px 25px",
+});
+
+const StyledFormControlLabel = styled(FormControlLabel)(({ theme }) => ({
+  width: "100%",
+  marginRight: 0,
+  marginLeft: 0,
+  "& .MuiFormControlLabel-label": {
+    width: "100%",
+    fontSize: "16px",
+    color: "#083A50 !important",
+    fontFamily: "'Nunito', 'Rubik', sans-serif",
+    fontStyle: "normal",
+    fontWeight: 700,
+    lineHeight: "19.6px",
+  },
+  "& .Mui-disabled": {
+    opacity: theme.palette.action.disabledOpacity,
+  },
+}));
+
+const StyledFooter = styled(Stack)({
+  flexDirection: "row",
+  justifyContent: "space-between",
+  alignItems: "center",
+  padding: "16px 8px 20px 13px",
+  borderTop: "1px solid #375F9A",
+  marginLeft: "12px",
+  marginRight: "12px",
+});
+
+type ColumnVisibilityModel = { [key: string]: boolean };
+
+type Props<C extends { hideable?: boolean }> = {
+  anchorEl: HTMLElement | null;
+  open: boolean;
+  columns: C[];
+  columnVisibilityModel: ColumnVisibilityModel;
+  sortAlphabetically?: boolean;
+  onColumnVisibilityModelChange: (model: ColumnVisibilityModel) => void;
+  onClose: () => void;
+  getColumnKey: (column: C) => string;
+  getColumnLabel: (column: C) => string;
+};
+
+/**
+ * A popper component that allows users to toggle the visibility of table columns.
+ * Supports non-hideable columns by disabling their checkboxes and labels.
+ *
+ * @template C - The type of the column objects
+ * @param {Props} props
+ * @returns {JSX.Element}
+ */
+const ColumnVisibilityPopper = <C extends { hideable?: boolean }>({
```

Comment:
Thanks for catching this, I wouldn't have noticed! This was caused by the placement of the popper being "bottom-start" which tried to prioritize opening the popper to the right instead of to the left, towards the table. I updated this to make the popper open towards the table and also disabled flipping in [85a18fa](https://github.com/CBIIT/crdc-datahub-ui/pull/476/commits/85a18fa6bf429ebccd54a935157a413aad9d9a1d).